// react-native-ios-utilities/lib/module/example_components/CardRowLabelDisplay.js
"use strict";

import * as React from 'react';
import { StyleSheet, View, Text } from 'react-native';
import { Colors } from "../misc/Colors.js";

/**
 * ```
 * ┌ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ 
 *   Label              Value │
 * └ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ 
 * ```
 */
import { jsx as _jsx, jsxs as _jsxs } from "react/jsx-runtime";
export function CardRowLabelDisplay(props) {
  return /*#__PURE__*/_jsxs(View, {
    style: styles.cardRowLabelDisplayContainer,
    children: [/*#__PURE__*/_jsx(Text, {
      style: styles.cardRowLabelDisplayLabelText,
      children: props.label ?? 'Current Value'
    }), /*#__PURE__*/_jsx(Text, {
      style: styles.cardRowLabelDisplayValueText,
      children: props.value ?? 'N/A'
    })]
  });
}
;
const styles = StyleSheet.create({
  cardRowLabelDisplayContainer: {
    flexDirection: 'row',
    marginTop: 12,
    paddingHorizontal: 12,
    paddingVertical: 5,
    backgroundColor: Colors.INDIGO[100],
    borderRadius: 10,
    alignItems: 'center',
    justifyContent: 'space-between'
  },
  cardRowLabelDisplayLabelText: {
    fontSize: 12,
    fontWeight: '500',
    color: Colors.PURPLE[1100],
    opacity: 0.75
  },
  cardRowLabelDisplayValueText: {
    fontSize: 12,
    fontWeight: '500',
    color: Colors.PURPLE[1100],
    opacity: 0.4
  }
});
