// react-native-ios-utilities/lib/commonjs/example_components/CardRowTextInput.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.CardRowTextInput = void 0;
var React = _interopRequireWildcard(require("react"));
var _reactNative = require("react-native");
var _Colors = require("../misc/Colors.js");
var _jsxRuntime = require("react/jsx-runtime");
function _interopRequireWildcard(e, t) { if ("function" == typeof WeakMap) var r = new WeakMap(), n = new WeakMap(); return (_interopRequireWildcard = function (e, t) { if (!t && e && e.__esModule) return e; var o, i, f = { __proto__: null, default: e }; if (null === e || "object" != typeof e && "function" != typeof e) return f; if (o = t ? n : r) { if (o.has(e)) return o.get(e); o.set(e, f); } for (const t in e) "default" !== t && {}.hasOwnProperty.call(e, t) && ((i = (o = Object.defineProperty) && Object.getOwnPropertyDescriptor(e, t)) && (i.get || i.set) ? o(f, t, i) : f[t] = e[t]); return f; })(e, t); }
class CardRowTextInput extends React.PureComponent {
  state = {
    textInput: ''
  };
  getText = () => {
    return this.state.textInput;
  };
  _handleOnChangeText = text => {
    this.setState({
      textInput: text
    });
  };
  render() {
    const props = this.props;
    return /*#__PURE__*/(0, _jsxRuntime.jsx)(_reactNative.TextInput, {
      style: styles.cardRowTextInput,
      onChangeText: this._handleOnChangeText,
      placeholder: props.placeholder,
      placeholderTextColor: _Colors.Colors.INDIGO[300]
    });
  }
}
exports.CardRowTextInput = CardRowTextInput;
;
const styles = _reactNative.StyleSheet.create({
  cardRowTextInput: {
    backgroundColor: _Colors.Colors.INDIGO[100],
    fontSize: 12,
    color: _Colors.Colors.INDIGO[900],
    paddingHorizontal: 12,
    paddingVertical: 7,
    borderRadius: 10,
    marginTop: 12
  }
});
