// react-native-ios-utilities/lib/commonjs/example_components/CardButton.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.CardButton = CardButton;
var React = _interopRequireWildcard(require("react"));
var _reactNative = require("react-native");
var _Colors = require("../misc/Colors.js");
var _jsxRuntime = require("react/jsx-runtime");
function _interopRequireWildcard(e, t) { if ("function" == typeof WeakMap) var r = new WeakMap(), n = new WeakMap(); return (_interopRequireWildcard = function (e, t) { if (!t && e && e.__esModule) return e; var o, i, f = { __proto__: null, default: e }; if (null === e || "object" != typeof e && "function" != typeof e) return f; if (o = t ? n : r) { if (o.has(e)) return o.get(e); o.set(e, f); } for (const t in e) "default" !== t && {}.hasOwnProperty.call(e, t) && ((i = (o = Object.defineProperty) && Object.getOwnPropertyDescriptor(e, t)) && (i.get || i.set) ? o(f, t, i) : f[t] = e[t]); return f; })(e, t); }
/**
 * ```
 * ┌─────────────────────────────┐
 * │ Title                       │
 * │ Subtitle                    │
 * └─────────────────────────────┘
 * ```
 */
function CardButton(props) {
  const buttonColor = props.buttonColor ?? _Colors.Colors.PURPLE.A200;
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_reactNative.TouchableOpacity, {
    style: [props.style, styles.cardButtonContainer, {
      backgroundColor: buttonColor
    }],
    onPress: props.onPress,
    children: /*#__PURE__*/(0, _jsxRuntime.jsxs)(React.Fragment, {
      children: [/*#__PURE__*/(0, _jsxRuntime.jsx)(_reactNative.Text, {
        style: styles.cardButtonTitleText,
        children: props.title
      }), /*#__PURE__*/(0, _jsxRuntime.jsx)(_reactNative.Text, {
        style: styles.cardButtonSubtitleText,
        children: props.subtitle
      })]
    })
  });
}
;
const styles = _reactNative.StyleSheet.create({
  cardButtonContainer: {
    paddingHorizontal: 12,
    paddingVertical: 8,
    borderRadius: 10,
    marginTop: 12
  },
  cardButtonTitleText: {
    color: 'white',
    fontSize: 14,
    fontWeight: '700'
  },
  cardButtonSubtitleText: {
    fontSize: 13,
    color: 'rgba(255,255,255,0.8)',
    fontWeight: '400'
  }
});
