// react-native-ios-utilities/lib/commonjs/example_components/CardRowSwitch.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.CardRowSwitch = CardRowSwitch;
var React = _interopRequireWildcard(require("react"));
var _reactNative = require("react-native");
var _Colors = require("../misc/Colors.js");
var _jsxRuntime = require("react/jsx-runtime");
function _interopRequireWildcard(e, t) { if ("function" == typeof WeakMap) var r = new WeakMap(), n = new WeakMap(); return (_interopRequireWildcard = function (e, t) { if (!t && e && e.__esModule) return e; var o, i, f = { __proto__: null, default: e }; if (null === e || "object" != typeof e && "function" != typeof e) return f; if (o = t ? n : r) { if (o.has(e)) return o.get(e); o.set(e, f); } for (const t in e) "default" !== t && {}.hasOwnProperty.call(e, t) && ((i = (o = Object.defineProperty) && Object.getOwnPropertyDescriptor(e, t)) && (i.get || i.set) ? o(f, t, i) : f[t] = e[t]); return f; })(e, t); }
/**
 * ```
 * ┌ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ┐
 *   Title               ┌──┬─┐   
 * │ Subtitle            └──┴─┘  │
 *  ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ 
 * ```
 */
function CardRowSwitch(props) {
  return /*#__PURE__*/(0, _jsxRuntime.jsxs)(_reactNative.View, {
    style: styles.cardRowSwitchContainer,
    children: [/*#__PURE__*/(0, _jsxRuntime.jsxs)(_reactNative.View, {
      style: styles.cardRowSwitchLabelContainer,
      children: [/*#__PURE__*/(0, _jsxRuntime.jsx)(_reactNative.Text, {
        style: styles.cardRowSwitchLabelText,
        children: props.title ?? 'title'
      }), /*#__PURE__*/(0, _jsxRuntime.jsx)(_reactNative.Text, {
        style: styles.cardRowSwitchSubtitleText,
        children: props.subtitle ?? 'Toggle the value'
      })]
    }), /*#__PURE__*/(0, _jsxRuntime.jsx)(_reactNative.Switch, {
      value: props.value ?? false,
      onValueChange: props.onValueChange,
      trackColor: {
        true: _Colors.Colors.PURPLE.A700,
        false: _Colors.Colors.PURPLE.A100
      }
    })]
  });
}
;
const styles = _reactNative.StyleSheet.create({
  cardRowSwitchContainer: {
    flexDirection: 'row',
    alignItems: 'center',
    marginTop: 12
  },
  cardRowSwitchLabelContainer: {
    flex: 1,
    marginRight: 10
  },
  cardRowSwitchLabelText: {
    fontSize: 12,
    fontWeight: '500',
    color: _Colors.Colors.PURPLE[1200]
  },
  cardRowSwitchSubtitleText: {
    fontSize: 12,
    opacity: 0.5,
    color: _Colors.Colors.PURPLE[1100]
  }
});
